src/libraries/System.Net.Http.WinHttpHandler/tests/FunctionalTests/WinHttpHandlerTest.cs

```diff
@@ -69,6 +71,79 @@ public async Task GetAsync_ConcurrentRead_ThrowsInvalidOperationException()
             await Task.WhenAll(tasks);
         }
 
+        [OuterLoop]
+        [ConditionalTheory(typeof(RemoteExecutor), nameof(RemoteExecutor.IsSupported))]
+        [MemberData(nameof(HttpVersions))]
+        public async Task SendAsync_ServerCertificateValidationCallback_CalledOnce(Version version)
+        {
+            await RemoteExecutor.Invoke(async (version) =>
+            {
+                AppContext.SetSwitch("System.Net.Http.UseWinHttpCertificateCaching", true);
+                int callbackCount = 0;
+                var handler = new WinHttpHandler()
+                {
+                    ServerCertificateValidationCallback = (_, _, _, _) =>
+                    {
+                        Interlocked.Increment(ref callbackCount);
+                        return true;
+                    }
+                };
+                using (var client = new HttpClient(handler))
+                {
+                    for (int i = 0; i < 5; i++)
+                    {
+                        var response = await client.SendAsync(new HttpRequestMessage(HttpMethod.Get, Configuration.Http.SecureRemoteEchoServer)
+                        {
+                            Version = Version.Parse(version)
+                        });
```

Comment:
This is not testing the HTTP/2 logic as we're silently downgrading to 1.1 since this endpoint only does 1.1.

Can we use a loopback server instead?